File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b_think/20260414_083459/python/002_word_frequency/solution.py
import collections
import re
import os

def solve():
    # Ensure the input directory and file exist to prevent crashes in local environments
    input_path = 'input/text.txt'
    if not os.path.exists(input_path):
        return

    with open(input_path, 'r', encoding='utf-8') as f:
        text = f.read()

    # Convert to lowercase and find all sequences of letters
    # Using re.findall with [a-z]+ ensures we only keep letters and strip punctuation
    words = re.findall(r'[a-z]+', text.lower())

    # Count frequencies
    counts = collections.Counter(words)

    # Sort by count descending (-x[1]), then by word ascending (x[0])
    sorted_words = sorted(counts.items(), key=lambda x: (-x[1], x[0]))

    # Output in the format word: count
    for word, count in sorted_words:
        print(f"{word}: {count}")

if __name__ == "__main__":
    solve()

File: codegen/results/v002/m3-max-48gb/ollama/gemma4_31b_think/20260414_083459/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * Solves the word frequency counting task.
 */
function solve() {
    const filePath = path.join('input', 'text.txt');
    
    let content: string;
    try {
        content = fs.readFileSync(filePath, 'utf8');
    } catch (error) {
        // The prompt implies the file exists, but we'll handle errors gracefully.
        return;
    }

    // Convert to lowercase
    const lowerContent = content.toLowerCase();

    // Extract words containing only letters
    // This strips all punctuation and keeps only alphabetical sequences
    const words = lowerContent.match(/[a-z]+/g);

    if (!words) {
        return;
    }

    // Count frequencies
    const frequencyMap = new Map<string, number>();
    for (const word of words) {
        frequencyMap.set(word, (frequencyMap.get(word) || 0) + 1);
    }

    // Convert Map to array for sorting
    const sortedWords = Array.from(frequencyMap.entries());

    // Sort logic:
    // 1. Count descending
    // 2. Alphabetical ascending (for ties)
    sortedWords.sort((a, b) => {
        const [wordA, countA] = a;
        const [wordB, countB] = b;

        if (countB !== countA) {
            return countB - countA;
        }
        return wordA.localeCompare(wordB);
    });

    // Output formatted results
    for (const [word, count] of sortedWords) {
        process.stdout.write(`${word}: ${count}\n`);
    }
}

// Run the solver
solve();